crates/cargo-bsp/tests/features_extension_integration_test.rs

```diff
@@ -0,0 +1,215 @@
+//! Integration tests for Cargo extension of BSP. The test in this module switches te environment,
+//! when adding extra tests remember to execute them sequentially.
+
+use bsp_server::Response;
+use bsp_types::requests::cargo_extension::*;
+use bsp_types::requests::Request;
+use cargo_bsp::utils::tests::create_feature_set_from_slices;
+use cargo_toml_builder::{types::Feature as TomlFeature, CargoToml};
+use serde_json::to_string;
+use std::collections::BTreeSet;
+use std::env::{current_dir, set_current_dir};
+use std::fs::File;
+use std::io::Write;
+use std::process::Command;
+use tempfile::tempdir;
+
+const TEST_REQUEST_ID: i32 = 123;
+const TEST_PROJECT_NAME: &str = "tmp_test_project";
+
+mod common;
+
+use common::{spawn_server_with_proper_life_time, Client};
+
+const F: [&str; 6] = ["f0", "f1", "f2", "f3", "f4", "f5"];
```

Comment:
może po prostu już w tym miejscu zrób ten BTreeSet<Feature> zamiast potem to mapować...